server/controllers/user.js

```diff
@@ -151,6 +150,83 @@ class User {
       { token },
     );
   }
+
+  /**
+   * This method gets a user profile
+   * @param {object} request
+   * @param {object} response
+   * @returns {object} the user data
+   */
+  static async profile(request, response) {
+    const data = await userRepo.getUserbyUsername(request.params.username);
+    if (data === null) {
+      return badHttpResponse(
+        response,
+        404,
+        'User not found',
+        'No data to show'
+      );
+    }
+    const {
+      id,
+      firstName,
+      lastName,
+      facebook,
+      google,
+      twitter,
+      bio,
+      imageUrl,
+      createdAt,
+      updatedAt,
+    } = data;
+    return goodHttpResponse(
+      response,
+      200,
+      'User found',
+      request.isAuthorized ? data : {
+        id,
+        firstName,
+        lastName,
+        facebook,
+        google,
+        twitter,
+        bio,
+        imageUrl,
+        createdAt,
+        updatedAt,
+      },
+    );
+  }
+
+  /**
+   *
+   * @param {object} request
+   * @param {object} response
+   * @returns {object} the response message to user
+   */
+  static async editProfile(request, response) {
+    if (!request.isAuthorized) {
+      return badHttpResponse(
+        response,
+        401,
+        'You are not permitted to complete this action',
+      );
+    }
+    if (request.body.images) {
+      const imageUrl = request.body.images[0];
+      request.body.imageUrl = imageUrl;
+    }
+    const updated = await userRepo.updateUser(request.body, request.params.username);
+    if (updated[0] === 1) {
```

Comment:
what's happening here? 🤔

Reply:
Sequelize returns the number of rows that were changed. In this case, I am expecting just one row.